platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmFeature.java

```diff
@@ -264,6 +266,25 @@ private Configuration createRuntimeElements(
         return runtimeElements;
     }
 
+    private Configuration createImplementationCompileElements(
+        RoleBasedConfigurationContainerInternal configurations,
+        PublishArtifact jarArtifact
+    ) {
+         String configName = getConfigurationName(JvmConstants.IMPLEMENTATION_COMPILE_ELEMENTS_CONFIGURATION_NAME);
+         return configurations.consumableUnlocked(configName, conf -> {
+            jvmLanguageUtilities.useDefaultTargetPlatformInference(conf, compileJava);
+            jvmPluginServices.configureAsImplementationCompileElements(conf);
+            capabilities.forEach(conf.getOutgoing()::capability);
+            conf.setDescription("Implementation compile elements for the '" + name + "' feature.");
+
+            conf.extendsFrom(implementation, compileOnly);
+
+            // Configure variants
+            addJarArtifactToConfiguration(conf, jarArtifact);
+            jvmPluginServices.configureClassesDirectoryVariant(conf, sourceSet);
```

Comment:
TODO: We should make this classes and resources. Unit tests already have classes and resources from the source set output, and just the classes is pretty much useless